cql/ManageRareAbnormality.cql

```diff
@@ -844,6 +948,40 @@ define SubsequentLowGradeCytologyReportsSinceInitialUnspecifiedHsilOrCin2InLast6
   else
     null
 
+
+define RecentNegativeHSILSurveillanceTests:
+  if (
+    (SubsequentLowGradeHistologyReportsSinceInitialUnspecifiedHsilOrCin2InLast31Years is not null) and
+    (SubsequentLowGradeCytologyReportsSinceInitialUnspecifiedHsilOrCin2InLast31Years is not null)) then
+     (
+       (
+        (SubsequentLowGradeHistologyReportsSinceInitialUnspecifiedHsilOrCin2InLast31Years) H 
+         with (SubsequentLowGradeCytologyReportsSinceInitialUnspecifiedHsilOrCin2InLast31Years) C
+           such that C.date same as H.date
+        ) union (
+         (SubsequentLowGradeHistologyReportsSinceInitialUnspecifiedHsilOrCin2InLast31Years) H 
+           with (NegativeHrHpvTests) N
+             such that N.date same as H.date 
+            )
+      ) R 
+      sort by date asc
+    else
+      null
+
+define FirstTwoNegativeHSILSurveillanceTests:
```

Comment:
I have a CQL VSCode extension throwing the following error:
```
Could not resolve call to operator Union with signature (tuple{riskTableInput:System.String,allConclusions:list<FHIR.CodeableConcept>,date:System.DateTime},tuple{riskTableInput:System.String,allConclusions:list<FHIR.CodeableConcept>,date:System.DateTime}).
```
I think you are only supposed to call a union on two lists, but the CQL-to-ELM translator just figures out how to union two singletons if they have the same signature.

Reply:
It seems like all you actually need from this expression is:
* A boolean determining whether or not at least 2 negative HSIL surveillance tests exist
* The date of the second test

I propose instead of having this expression returning the two tests themselves, you could have it return the date of the second test if the count is >= 2, or else return null. That way you could say "if FirstTwoNegativeHSILSurveillanceTests is not null" every time you are trying to evaluate whether the two tests have happened yet, and then when you need the date, you just call this expression.

Reply:
Will refactor to a boolean for HasTwoNegativeHsilSurveillanceTests and date of second test SecondNegativeHsilSurveillanceTestDate

Reply:
Great! This removes the warning that was thrown by trying to `union` two singletons, and I think it also makes the section easier to read.